Content.Client/GameObjects/Components/Mobs/SpeciesVisualizer2D.cs

```diff
@@ -11,19 +16,51 @@ public override void OnChangeData(AppearanceComponent component)
         {
             base.OnChangeData(component);
 
-            var sprite = component.Owner.GetComponent<ISpriteComponent>();
             if (component.TryGetData<SharedSpeciesComponent.MobState>(SharedSpeciesComponent.MobVisuals.RotationState, out var state))
             {
                 switch (state)
                 {
-                    case SharedSpeciesComponent.MobState.Stand:
-                        sprite.Rotation = 0;
+                    case SharedSpeciesComponent.MobState.Standing:
+                        SetRotation(component, 0);
                         break;
                     case SharedSpeciesComponent.MobState.Down:
-                        sprite.Rotation = Angle.FromDegrees(90);
+                        SetRotation(component, Angle.FromDegrees(90));
                         break;
                 }
             }
         }
+
+        public void SetRotation(AppearanceComponent component, Angle rotation)
```

Comment:
Shouldn't you add an input field to this function for rotation duration? I see that you have Length = TimeSpan.FromSeconds(0.125).

Reply:
uhh, that is the length of the rotation animation, not the length of the rotation itself

Reply:
You know, I'll make it private actually.